src/tasks/DockerBuildTaskProvider.ts

```diff
@@ -0,0 +1,116 @@
+/*---------------------------------------------------------------------------------------------
+ *  Copyright (c) Microsoft Corporation. All rights reserved.
+ *  Licensed under the MIT License. See LICENSE.md in the project root for license information.
+ *--------------------------------------------------------------------------------------------*/
+
+import * as fse from 'fs-extra';
+import { CancellationToken, ProviderResult, ShellExecution, ShellQuotedString, Task, TaskProvider, WorkspaceFolder } from 'vscode';
+import { callWithTelemetryAndErrorHandling, IActionContext } from 'vscode-azureextensionui';
+import { DockerPlatform, getPlatform } from '../debugging/DockerPlatformHelper';
+import { cloneObject } from '../utils/cloneObject';
+import { CommandLineBuilder } from '../utils/commandLineBuilder';
+import { resolveFilePath } from '../utils/resolveFilePath';
+import { DockerBuildOptions } from './DockerBuildTaskDefinitionBase';
+import { NetCoreBuildTaskDefinition } from './netcore/NetCoreTaskHelper';
+import { NodeBuildTaskDefinition } from './node/NodeTaskHelper';
+import { DockerBuildTaskContext, TaskHelper } from './TaskHelper';
+
+export interface DockerBuildTaskDefinition extends NetCoreBuildTaskDefinition, NodeBuildTaskDefinition {
+    label?: string;
+    dependsOn?: string[];
+    platform?: DockerPlatform;
+}
+
+export interface DockerBuildTask extends Task {
+    definition: DockerBuildTaskDefinition;
+}
+
+export class DockerBuildTaskProvider implements TaskProvider {
+    constructor(private readonly helpers: { [key in DockerPlatform]: TaskHelper }) { }
+
+    public provideTasks(token?: CancellationToken): ProviderResult<Task[]> {
+        return []; // Intentionally empty, so that resolveTask gets used
+    }
+
+    public resolveTask(task: DockerBuildTask, token?: CancellationToken): ProviderResult<Task> {
+        const taskPlatform = getPlatform(task.definition);
+        return callWithTelemetryAndErrorHandling(
+            `docker-build-resolve/${taskPlatform || 'unknown'}`,

```

Comment:
We don't usually have dynamic event names like this. It's simpler to have one event name that you only have to register once with gdpr and then stick `taskPlatform` as a property on `context.telemetry.properties`

Reply:
Yeah, this was before I'd figured out IActionContext and attached the platform as a property of it

Reply:
Resolved in: https://github.com/microsoft/vscode-docker/pull/1278